### ISSUE TITLE
Investigate Parallel Builds for MSVC

### ISSUE THREAD

**Issue by OwenArnold** [scipp/scipp#2078]
Slow build times for visual studio might result in missing parallel build flags.

The pipelines repo calls `build_cpp.py` in scipp. Parallel build should happen via `cmake --build . -jN` where N is number of cores to use https://github.com/scipp/scipp/blob/main/tools/build_cpp.py#L41
https://github.com/scipp/scipp/blob/main/tools/build_cpp.py#L99
Two suspect things. (1) if you hard code N, to a smaller number than the default (I think default is 4 for the azure provided vms and I used 2) there is no change in build time. (2) In the logs on azure I see no indication that msbuild is attempting to run a parallel build.

**Comment by Tom-Willemsen** [comment]
TL;DR version: on windows/MSVC the `cmake --build --parallel` flag has the counterintuitive effect of *disabling* useful parallelism.

---

There are 2 levels of parallelism available in the MSVC world...
- `-mN` as a flag on MSBuild, which allows MSBuild to build N different projects in parallel
- `/p:CL_MPCount=N` which allows the actual cl.exe compiler to compile different files within a project in parallel.

In addition to the above, there is also a [relatively new flag `UseMultiToolTask`](https://devblogs.microsoft.com/cppblog/improved-parallelism-in-msbuild/). This attempts to balance the two levels of parallelism described above (to try to limit either under or oversubscription of resources).

By default `cmake -jN` changes `-m` (project parallelism), and sets `/p:CL_MPCount` (compiler parallelism) to 1 ([ref 1](https://discourse.cmake.org/t/parallel-does-not-really-enable-parallel-compiles-with-msbuild/964/10), [ref 2](https://gitlab.kitware.com/cmake/cmake/-/issues/20564)). 

This isn't useful for us as we're building a single project - we would much rather leave `CL_MPCount` at it's default (number of cores in machine).



On my dev machine (2 cores, 4 threads) I get the following results (after running the setup command below each time):

```
git clean -fqdx ..
cmake -A x64 -G "Visual Studio 16 2019" -DCMAKE_BUILD_TYPE=Debug -DPYTHON_EXECUTABLE=C:\Users\<username>\Miniconda3\envs\scipp-developer\python.exe -DCMAKE_INSTALL_PREFIX=..\install -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=OFF -DDYNAMIC_LIB=ON -DDISABLE_MULTI_THREADING=ON ..
```

| command | time | Underlying MSBuild parallelism flags |
| - | - | - |
| `time cmake --build .` | 7:35.96 | (none, i.e. default values) |
| `time cmake --build . -j1` | 13:59.36 | `/p:CL_MPCount=1` `-m1` |
| `time cmake --build . -j2` | 14:32.46 | `/p:CL_MPCount=1` `-m2` |
| `time cmake --build . -j4` | 14:13.60 | `/p:CL_MPCount=1` `-m4` |
| `time cmake --build . -- -p:UseMultiToolTask=true -p:CL_MPCount=4` | 8:03.99 | `-p:UseMultiToolTask=true` `-p:CL_MPCount=4` |

**Comment by OwenArnold** [comment]
@Tom-Willemsen the same should be done for scipp-neutron.
